lib/datadog/lambda/utils/extension.rb

```diff
@@ -26,16 +28,42 @@ def self.extension_running?
     end
 
     def self.check_extension_running
-      return false unless File.exist?(EXTENSION_PATH)
+      File.exist?(EXTENSION_PATH)
+    end
+
+    def self.send_start_invocation_request(event:)
+      return unless extension_running?
 
-      begin
-        Net::HTTP.get(EXTENSION_CHECK_URI)
-      rescue StandardError => e
-        Datadog::Utils.logger.debug "extension is not running, returned with error #{e}"
-        return false
+      response = Net::HTTP.post(START_INVOCATION_URI, event.to_json, request_headers)
+
+      trace_digest = Tracing::Propagation::HTTP.extract(response)
+      # Only continue trace from a new one if it exist, or else,
+      # it will create a new trace, which is not ideal here.
+      Tracing.continue_trace!(trace_digest) if trace_digest
+    rescue StandardError => e
+      Datadog::Utils.logger.debug "failed on start invocation request to extension: #{e}"
+    end
+
+    def self.send_end_invocation_request(response:)
+      return unless extension_running?
+
+      request = Net::HTTP::Post.new(END_INVOCATION_URI)
+      request.body = response.to_json
+      request[Datadog::Transport::Ext::HTTP::HEADER_DD_INTERNAL_UNTRACED_REQUEST] = 1
+
+      trace = Datadog::Tracing.active_trace
+      Tracing::Propagation::HTTP.inject!(trace, request)
+      Net::HTTP.start(END_INVOCATION_URI.host, END_INVOCATION_URI.port) do |http|
+        http.request(request)
       end
+    rescue StandardError => e
+      Datadog::Utils.logger.debug "failed on end invocation request to extension: #{e}"
+    end
 
-      true
+    def self.request_headers
+      {
+        Datadog::Transport::Ext::HTTP::HEADER_DD_INTERNAL_UNTRACED_REQUEST.to_sym => 'true'
```

Comment:
can you clarify what this is? why would we add headers if its untraced?

Reply:
https://stackoverflow.com/questions/4062929/what-does-the-ruby-method-to-sym-do not sure I understand

Reply:
https://ruby-doc.org/core-2.5.1/Symbol.html

Reply:
Sometimes, headers require to be symbols. I'd suggest reading the link above to learn more about Symbols!

Reply:
The headers are to mark the request as untraced.
We add them because this what the extension requires in order to correlate traces and their propagation.
This are calls from our library to the layer. We don't want them to be showing in the users tracers.

Reply:
I agree that it's good to add a comment saying that this header does.